bionic/protocols.py

```diff
@@ -197,6 +198,39 @@ def __repr__(self):
         return f"{self.__class__.__name__}(...)"
 
 
+class JsonProtocol(BaseProtocol):
+    """
+    Decorator indicating that an entity's values can be serialized to the
+    ``JSON`` format.
+    """
+
+    def get_fixed_file_extension(self):
+        return "json"
+
+    def validate(self, value):
+        if value is None:
+            return
+        if isinstance(value, list):
+            for elem in value:
+                self.validate(elem)
+            return
+        if isinstance(value, dict):
+            for key, elem in value.items():
+                assert isinstance(key, str)
+                self.validate(elem)
+            return
+
+        assert isinstance(value, (int, float, str, bool))
+
+    def write(self, value, path):
+        with path.open("w", encoding="utf-8") as file_:
+            json.dump(value, file_, ensure_ascii=False)
+
+    def read(self, path):
+        with path.open("r") as file_:
```

Comment:
Oh I probably copied the read write methods from pickle protocol and rewrote the write method. Given that we don't give it an encoding for other protocols (only yaml to be fair since it's the only one opening it in text mode), I'll keep it consistent and remove the encoding for write.

Reply:
Hmm... if you don't specify an encoding, what encoding does it use?

Reply:
We talked more about this off-GitHub and decided to add encoding to all text files explicitly.